core/src/main/scala/org/bitcoins/core/wallet/utxo/UTXOSpendingInfo.scala

```diff
@@ -83,6 +83,18 @@ object ConditionalPath {
 
   val nonNestedTrue: ConditionalPath = ConditionTrue(NoConditionsLeft)
   val nonNestedFalse: ConditionalPath = ConditionFalse(NoConditionsLeft)
+
+  def fromBranch(branch: Vector[Boolean]): ConditionalPath = {
```

Comment:
tailrec with accumulator?

Reply:
sadly no again because the last call in this method is to `ConditionalPath` constructors. Essentially the below is equivalent to:
```
val condition = fromBranch(branch.tail)
ConditionTrue(condition)
```